perceval/components/abstract_processor.py

```diff
@@ -238,21 +239,33 @@ def add(self, mode_mapping, component, keep_port=True):
         >>> p.add([2,5], BS())  # Modes (2, 5) of the processor's output connected to (0, 1) of the added beam splitter
         >>> p.add({2:0, 5:1}, BS())  # Same as above
         """
-        if self._n_moi is None:
-            self._n_moi = component.m + mode_mapping if isinstance(mode_mapping, int) else max(mode_mapping) + 1
+        if self._n_moi == 0:
+            self.m = component.m + mode_mapping if isinstance(mode_mapping, int) else max(mode_mapping) + 1
             get_logger().debug(f"Number of modes of interest defaulted to {self._n_moi} in processor {self.name}",
-                         channel.general)
+                               channel.general)
 
         connector = ModeConnector(self, component, mode_mapping)
         if isinstance(component, AProcessor):
             self._compose_processor(connector, component, keep_port)
+        elif isinstance(component, IDetector):
+            self._add_detector(mode_mapping, component)
         elif isinstance(component, AComponent):
             self._add_component(connector.resolve(), component)
         else:
             raise RuntimeError(f"Cannot add {type(component)} object to a Processor")
         self._circuit_changed()
         return self
 
+    def _add_detector(self, mode: int, detector: IDetector):
+        if not isinstance(mode, int):
+            raise TypeError(f"When adding a detector, the mode number must be an integer (got {type(mode)})")
```

Comment:
If it's a list of int, can we loop over the values, as iterable are supposed to be supported in `add`?